scripts/docgen.js

```diff
@@ -118,164 +159,183 @@ const generateMarkdown = ({ contractFile, templateFile, outputFile }) => {
+                });
+
+                // Push the param and continue loop
+                natspec.params.push(param);
               }
 
-              // Set natspec return including additional natspec return lines
-              let param = contractFileArray[returnLineIndex].split(" @return ")[1];
-              additionalReturnLineIndexes.forEach((index) => {
-                param += contractFileArray[index].split("///")[1];
-              });
-
-              // Push the return and continue loop
-              natspec.returns.push(param);
+              // Incremenent the param line
+              paramLineIndex += 1;
             }
+          }
 
-            // Incremenent the return line
-            returnLineIndex += 1;
+          // Check whether the method has returns
+          if (method.returnParameters) {
+            // Set the initial value for the return line index
+            let returnLineIndex = noticeLineIndex + 1;
+
+            // Get the natspec return line index for each return
+            while (
+              contractFileArray[returnLineIndex] &&
+              contractFileArray[returnLineIndex].includes("///") &&
+              natspec.returns.length !== method.returnParameters.length
+            ) {
+              // Check whether the current line is a valid return line
+              if (contractFileArray[returnLineIndex].includes(" @return ")) {
+                // Get additional natspec return lines
+                const additionalReturnLineIndexes = [];
+                let additionalReturnLineIndex = returnLineIndex + 1;
+                while (isValidAdditionalLine(additionalReturnLineIndex)) {
+                  additionalReturnLineIndexes.push(additionalReturnLineIndex);
+                  additionalReturnLineIndex += 1;
+                }
+
+                // Set natspec return including additional natspec return lines
+                let param = contractFileArray[returnLineIndex].split(" @return ")[1];
+                additionalReturnLineIndexes.forEach((index) => {
+                  param += contractFileArray[index].split("///")[1];
+                });
+
+                // Push the return and continue loop
+                natspec.returns.push(param);
+              }
+
+              // Incremenent the return line
+              returnLineIndex += 1;
+            }
           }
+        } else {
+          // Log warning for any methods without a valid notice line
+          // console.error(`Warning: ${method.name} is missing a natspec @notice`);
+          // foundError = true;
         }
-      } else {
-        // Log warning for any methods without a valid notice line
-        console.error(`Warning: ${method.name} is missing a natspec @notice`);
-        foundError = true;
-      }
 
-      // Push the method and append natspec
-      methods.push({ ...method, natspec });
-    });
+        methods.push({ ...method, natspec });
+      });
+  }
```

Comment:
Is there any way to generalize all this repetitive logic? Helper functions? Another level of abstraction?